errors_test.go

```diff
@@ -72,3 +73,36 @@ func TestNewTimeErrorErrorNil(t *testing.T) {
 	te := NewTimeError(nil)
 	r.Equal("", te.Error())
 }
+
+func TestPrefixErrorRetrieveError(t *testing.T) {
+	var r = require.New(t)
+
+	var err *oauth2.RetrieveError
+	if err := PrefixError("failed to get access token", err); err != nil {
+		r.Equal("ErrInvalidCredentials: failed to get access token: <nil>", err.Error())
```

Comment:
As I understand based on `LastError` doc, it should return `ErrInvalidCredentials`, not error whose string starts with `ErrInvalidCredentials`. Otherwise, following code wouldn't work which is the main purpose of returning specific error types:
```go
err, t := provider.LastError()
if _, ok := err.(ErrInvalidCredentials); ok {
    // display "Invalid credentials provided" on the UI
}
```
Am I misunderstanding?

Reply:
This test is testing the `PrefixError` function which prefixes errors with a string, it's one half of the operation. Those prefixes are later processed in the `run` function by the `NewTimeError` function, during the other half of the operation. This is done so we can catch broker and provider errors without knowing that context from `run`.

You can observe the entire operation from [this](https://github.com/hashicorp/hcp-scada-provider/pull/33#pullrequestreview-1117425166) test in `provider_test.go`.

Reply:
Note that broker will also call `PrefixError` for the situation over there.